lib/cloud_controller/metrics/prometheus_updater.rb

```diff
@@ -4,11 +4,30 @@ module VCAP::CloudController::Metrics
   class PrometheusUpdater
     def initialize(registry=Prometheus::Client.registry)
       @registry = registry
+
+      # Register all metrics, to initialize them for discoverability
+      @registry.gauge(:cc_job_queues_length_total, docstring: 'Job queues length of worker processes', labels: [:queue]) unless @registry.exist?(:cc_job_queues_length_total)
+      @registry.gauge(:cc_failed_jobs_total, docstring: 'Number of failed jobs of worker processes', labels: [:queue]) unless @registry.exist?(:cc_failed_jobs_total)
```

Comment:
Good question. Currently we read the failed jobs from the DB. As old jobs get cleaned up by a background job, a lower number will be reported. I think this also makes more sense than keeping an ever increasing counter, which will loose quickly any meaning, except people track the rate with which the counter increases.
Another idea would be to keep the number of failed jobs in memory, so that when a worker gets restarted it starts at 0 again. (then we need a prom endpoint on worker VMs though).